plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -492,12 +498,46 @@ private IcebergTableHandle tableHandleForSnapshot(
                 Optional.ofNullable(tableProperties.get(TableProperties.DEFAULT_NAME_MAPPING)),
                 table.location(),
                 table.properties(),
+                getTablePartitioning(session, table),
                 false,
                 Optional.empty(),
                 ImmutableSet.of(),
                 Optional.of(false));
     }
 
+    private Optional<IcebergTablePartitioning> getTablePartitioning(ConnectorSession session, Table icebergTable)
+    {
+        // todo table partitioning should be allowed when there are multiple specs, but only one is active, or if the specs are compatible
+        BucketExecutionMode bucketExecutionMode = getBucketExecutionMode(session);
+        if (bucketExecutionMode == BucketExecutionMode.NEVER || icebergTable.specs().size() != 1) {
+            return Optional.empty();
+        }
+        PartitionSpec partitionSpec = icebergTable.spec();
+        if (partitionSpec.fields().isEmpty()) {
+            return Optional.empty();
+        }
+
+        Schema schema = icebergTable.schema();
+
+        IcebergPartitioningHandle icebergPartitioningHandle = IcebergPartitioningHandle.create(partitionSpec, typeManager, List.of());
+
+        // In auto mode, we only use bucketing if the estimated number of buckets is less than or equal to the number of workers.
```

Comment:
It's actually opposite. There has to be sufficient buckets to fully utilize the tasks, see `DetermineTableScanNodePartitioning`
```
        return Result.ofPlanNode(node
                .withUseConnectorNodePartitioning((double) numberOfBuckets / numberOfTasks >= getTableScanNodePartitioningMinBucketToTaskRatio(context.getSession())))
```

Reply:
This was a WIP and this part got rewritten in the final code